hibernate-core/src/main/java/org/hibernate/loader/hql/QueryLoader.java

```diff
@@ -450,9 +450,16 @@ protected Object getResultColumnOrRow(
 		if ( hasScalars ) {
 			String[][] scalarColumns = scalarColumnNames;
 			int queryCols = queryReturnTypes.length;
+			int managedTypeIndex = 0;
 			resultRow = new Object[queryCols];
 			for ( int i = 0; i < queryCols; i++ ) {
-				resultRow[i] = queryReturnTypes[i].nullSafeGet( rs, scalarColumns[i], session, null );
+				// The scalar column names can only be empty if this is a managed type which was already materialized by it's persister
```

Comment:
Minor nitpicking: `s/it's/its/`.

Maybe I would have gone for `When the column is a managed type, the scalar column names array is empty and we have to get the result from the rows array.`.